middleware/compress.go

```diff
@@ -0,0 +1,186 @@
+package middleware
+
+import (
+	"compress/flate"
+	"compress/gzip"
```

Comment:
I'm thinking of exposing the writers via API, so users could inject 
https://github.com/klauspost/compress or https://github.com/klauspost/pgzip